src/hashutil.h

```diff
@@ -40,22 +40,54 @@ class HashUtil {
   static std::string MD5Hash(const char *inbuf, size_t in_length);
   static std::string SHA1Hash(const char *inbuf, size_t in_length);
 
-  // See Martin Dietzfelbinger, "Universal hashing and k-wise independent random
-  // variables via integer arithmetic without primes".
-  static uint64_t TwoIndependentMultiplyShift(uint64_t key) {
-    const uint64_t SEED[4] = {0x818c3f78ull, 0x672f4a3aull, 0xabd04d69ull,
-                              0x12b51f95ull};
-    const unsigned __int128 m =
-        *reinterpret_cast<const unsigned __int128 *>(&SEED[0]);
-    const unsigned __int128 a =
-        *reinterpret_cast<const unsigned __int128 *>(&SEED[2]);
-    return (a + m * key) >> 64;
-  }
-
  private:
   HashUtil();
 };
 
-}  // namespace cuckoofilter
+// See Martin Dietzfelbinger, "Universal hashing and k-wise independent random
+// variables via integer arithmetic without primes".
+class TwoIndependentMultiplyShift {
+  unsigned __int128 multiply_, add_;
+
+ public:
+  TwoIndependentMultiplyShift() {
+    ::std::random_device random;
+    for (auto v : {&multiply_, &add_}) {
+      *v = random();
+      for (int i = 1; i <= 4; ++i) {
+        *v = *v << 32;
+        *v |= random();
+      }
+    }
+  }
+
+  uint64_t operator()(uint64_t key) const {
+    return (add_ + multiply_ * static_cast<decltype(multiply_)>(key)) >> 64;
+  }
+};
+
+// See Patrascu and Thorup's "The Power of Simple Tabulation Hashing"
+class SimpleTabulation {
+  uint64_t tables_[sizeof(uint64_t)][1 << CHAR_BIT];
+
+ public:
+  SimpleTabulation() {
+    ::std::random_device random;
+    for (unsigned i = 0; i < sizeof(uint64_t); ++i) {
+      for (int j = 0; j < (1 << CHAR_BIT); ++j) {
+        tables_[i][j] = random() | ((static_cast<uint64_t>(random())) << 32);
+      }
+    }
+  }
+
+  uint64_t operator()(uint64_t key) const {
+    uint64_t result = 0;
+    for (unsigned i = 0; i < sizeof(key); ++i) {
+      result ^= tables_[i][reinterpret_cast<uint8_t *>(&key)[i]];
+    }
+    return result;
+  }
+};
+}
 
-#endif  // CUCKOO_FILTER_HASHUTIL_H_
+#endif  // #ifndef _HASHUTIL_H_
```

Comment:
emm, here `// CUCKOO_FILTER_HASHUTIL_H_`  is Google C++ style convention. Any reason to change?